components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx

```diff
@@ -22,19 +22,20 @@ interface Props extends PropsFromRedux {
 }
 
 export default class SidebarBaseChannel extends React.PureComponent<Props> {
-    handleLeavePublicChannel = () => {
+    handleLeavePublicChannel = (callback: () => void) => {
         this.props.actions.leaveChannel(this.props.channel.id);
         trackEvent('ui', 'ui_public_channel_x_button_clicked');
+        callback();
     }
 
-    handleLeavePrivateChannel = () => {
+    handleLeavePrivateChannel = (callback: () => void) => {
         this.props.actions.openModal({modalId: ModalIdentifiers.LEAVE_PRIVATE_CHANNEL_MODAL, dialogType: LeavePrivateChannelModal, dialogProps: {channel: this.props.channel}});
         trackEvent('ui', 'ui_private_channel_x_button_clicked');
+        callback();
     }
 
     getCloseHandler = () => {
```

Comment:
Could we add a unit test for this component's instance to ensure getCloseHandler works well?

Reply:
Hello @codedsun . Do you need any help on this?

Reply:
@AGMETEOR - Yes I am a little confused to where to write the unit test and how?

Reply:
This would be the file with the file https://github.com/mattermost/mattermost-webapp/blob/master/components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.test.tsx where to put the test. It's just next to the file where you made the changes.

See example here on how to test a method inside a class component https://stackoverflow.com/a/41196670/7723621

Reply:
> This would be the file with the file https://github.com/mattermost/mattermost-webapp/blob/master/components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.test.tsx where to put the test. It's just next to the file where you made the changes.
> 
> See example here on how to test a method inside a class component https://stackoverflow.com/a/41196670/7723621

Let me know if you need more help.

Reply:
@AGMETEOR Let me try today and if I can't do it, will ping you and connect for a call. Thanks

Reply:
@AGMETEOR Done 💯